### ISSUE TITLE
Locating MD Files is not  working

### ISSUE THREAD

**Issue by Litt7e** [daeh/zotero-markdb-connect#30]
The  **_Choose_** botton of Folder Containg Markdown Reading Note is not working.  Nothing show up after I click the  **_Choose_** botton. 
And I copy the _**file path**_ of md notes in the text box, but the pulug in still can not find md nites.

**Comment by daeh** [comment]
thanks for bringing to my attention. I introduced a bug in that button when I was working on something else. Upgrade to https://github.com/daeh/zotero-markdb-connect/releases/tag/v0.0.20 and let me know if that resolves your issue or not

**Comment by Litt7e** [comment]
The **Choose** botton of Folder Containg Markdown Reading Note is working now. But it seems that snyc Tags is not working, I am sure the **file path** is  correct.
![image](https://user-images.githubusercontent.com/44261353/204128344-dd9d83ab-2bf7-49c9-b484-451368006907.png)

**Comment by daeh** [comment]
I'll need more info to help with that. What's the path, what's the filter filter, and what's an example markdown note that you expect to be matched?

**Comment by Litt7e** [comment]
The path is the Export Path of BibNotes Formatter. But the Note Title is not {{citekey}}, I change it to {{title}}, I don't know whether it is the problem.
![image](https://user-images.githubusercontent.com/44261353/204129603-4ca74189-3445-4f67-a48c-401871da7b1e.png)
![image](https://user-images.githubusercontent.com/44261353/204129641-900b50a6-3f23-4de2-8381-04a51d249fe6.png)

**Comment by daeh** [comment]
So it's 2 things. 

1) your custom file filter regex is looking for files that begin with `@`. To find your reading notes use `^.+\.md$`.

2) if you're using a custom file filter, you'll need to specify how the plugin should match the markdown files to zotero items. Assuming that you're using BBT, then the citekey needs to be in the metadata, like in the example MD files of the readme:

<img width="859" alt="CleanShot 2022-11-27 at 05 14 03@2x" src="https://user-images.githubusercontent.com/12652166/204129922-a0ff6203-82bb-4a4f-9a30-cabeec4b79c3.png">

**Comment by daeh** [comment]
The literature note content template I use is 

```
---
citekey: {{citekey}}
year: {{year}}
authors: {{authorString}}
title: "{{title}}"
aliases: ["@{{citekey}}", "@{{citekey}} {{~#each entry.author~}} {{~#if @last}} ({{~family~}}) {{~/if~}} {{~/each}} {{title}}"]
authorFirst: {{entry.author.[0].family}}, {{entry.author.[0].given}}
authorLast: {{#each entry.author ~}} {{~#if @last~}} {{~family}}, {{given~}} {{~/if~}} {{~/each}}
publication: {{containerTitle}}
doi: {{DOI}}
zoterouri: {{zoteroSelectURI}}
obsidiankey: "@{{citekey}}"
tags: 
- readingNote
---

**{{title}}**
{{authorString}}
({{year}}) {{~#if containerTitle}} {{containerTitle~}} {{~else~}} {{~/if}}
[@{{citekey}}]({{zoteroSelectURI}})
```

(with the title template `@{{citekey}}`)

**Comment by Litt7e** [comment]
My markdb-connect config is like this:
![image](https://user-images.githubusercontent.com/44261353/204130842-f9039da3-c175-49f6-a415-2c1f03f6ad44.png)
My literature note content template I use is like this
![image](https://user-images.githubusercontent.com/44261353/204130886-d317d85e-1ec9-40b6-b16b-23fa0db1d58b.png)
The error report are:
![image](https://user-images.githubusercontent.com/44261353/204130779-1c37715c-b0ee-4d60-9f7b-18b8484ea2ce.png)
![image](https://user-images.githubusercontent.com/44261353/204130803-3e6fb7b4-a04c-42fb-953a-24800c4db6c6.png)
![image](https://user-images.githubusercontent.com/44261353/204130825-03680d8a-cbc7-478f-b8cc-8da7c376bfea.png)

 Could you please tell me how to set the yaml keyword

**Comment by daeh** [comment]
put this in the "Literature note content template"

```
---
CiteKey: {{citekey}}
---
```

<img width="776" alt="CleanShot 2022-11-27 at 12 17 45@2x" src="https://user-images.githubusercontent.com/12652166/204150113-41e7533a-8f7f-451c-a76c-8386f3f85ce1.png">


so that the very top of the MD file begins with

```
---
CiteKey: sonImplantableNeuralStimulator2020
---
```

and put `CiteKey`  in the yaml keyword field of the plugin

**Comment by Litt7e** [comment]
I have tried to manually modify the Metadata in two ways, it seems that they are both ineffective. And I have made so many notes that I can't bear the consequences of completely regenerating the files in the new format. I'm afraid I have to give up using this plug-in regretfully. Thank you for your help. If the subsequent update allows me to use this plug-in without regenerating notes, I will start using it again.


![image](https://user-images.githubusercontent.com/44261353/204183267-32a9b789-6deb-4b83-a058-5b0b6bbfab51.png)
![image](https://user-images.githubusercontent.com/44261353/204183529-58d25c15-059c-4c54-9f4e-2bb4bdf48244.png)

**Comment by daeh** [comment]
sure. It looks like you're not including the `---` above and below `CiteKey: sonImplantableNeuralStimulator2020`. Markdown metadata (aka frontmatter) is defined by having three dashes at the head of the file and another three dashes at the end the frontmatter. The example MD file in the readme of this plugin gives an example; you can also see the obsidian guide:
https://help.obsidian.md/Advanced+topics/YAML+front+matter

Again, it should work if that file begins with

```
---
CiteKey: sonImplantableNeuralStimulator2020
---

# An Implantable....
```

**Comment by daeh** [comment]
closed by ce46c9b3496c44a33d99288f6d311ee5389a9978
